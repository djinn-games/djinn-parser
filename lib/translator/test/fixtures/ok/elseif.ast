{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "elseifs",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "IfSentence",
        "if": {
          "condition": {
            "type": "Literal",
            "line": 3,
            "dataType": "bool",
            "value": false
          },
          "consequent": [
            {
              "type": "ExpressionSentence",
              "expression": {
                "type": "Literal",
                "line": 4,
                "dataType": "int",
                "value": 0
              },
              "line": 4
            }
          ],
          "alternates": [
            [
              {
                "type": "IfSentence",
                "if": {
                  "condition": {
                    "type": "Literal",
                    "line": 5,
                    "dataType": "bool",
                    "value": false
                  },
                  "consequent": [
                    {
                      "type": "ExpressionSentence",
                      "expression": {
                        "type": "Literal",
                        "line": 6,
                        "dataType": "int",
                        "value": 1
                      },
                      "line": 6
                    }
                  ],
                  "alternates": []
                },
                "line": 5
              }
            ],
            [
              {
                "type": "IfSentence",
                "if": {
                  "condition": {
                    "type": "Literal",
                    "line": 7,
                    "dataType": "bool",
                    "value": true
                  },
                  "consequent": [
                    {
                      "type": "ExpressionSentence",
                      "expression": {
                        "type": "Literal",
                        "line": 8,
                        "dataType": "int",
                        "value": 2
                      },
                      "line": 8
                    }
                  ],
                  "alternates": []
                },
                "line": 7
              }
            ]
          ],
          "line": 3
        },
        "line": 3
      }
    ],
    "line": 2
  },
  "line": 1
}