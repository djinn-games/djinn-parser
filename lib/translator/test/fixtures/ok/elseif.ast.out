{
  "type": "Program",
  "body": [
    {
      "type": "IfStatement",
      "test": {
        "type": "Literal",
        "value": false,
        "raw": "false"
      },
      "consequent": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "Literal",
              "value": 0,
              "raw": "0"
            }
          }
        ]
      },
      "alternate": {
        "type": "IfStatement",
        "test": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        },
        "consequent": {
          "type": "BlockStatement",
          "body": [
            {
              "type": "ExpressionStatement",
              "expression": {
                "type": "Literal",
                "value": 1,
                "raw": "1"
              }
            }
          ]
        },
        "alternate": {
          "type": "IfStatement",
          "test": {
            "type": "Literal",
            "value": true,
            "raw": "true"
          },
          "consequent": {
            "type": "BlockStatement",
            "body": [
              {
                "type": "ExpressionStatement",
                "expression": {
                  "type": "Literal",
                  "value": 2,
                  "raw": "2"
                }
              }
            ]
          },
          "alternate": {
            "type": "BlockStatement",
            "body": []
          }
        }
      }
    },
    {
      "type": "IfStatement",
      "test": {
        "type": "Literal",
        "value": false,
        "raw": "false"
      },
      "consequent": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "Literal",
              "value": 3,
              "raw": "3"
            }
          }
        ]
      },
      "alternate": {
        "type": "IfStatement",
        "test": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        },
        "consequent": {
          "type": "BlockStatement",
          "body": [
            {
              "type": "ExpressionStatement",
              "expression": {
                "type": "Literal",
                "value": 4,
                "raw": "4"
              }
            }
          ]
        },
        "alternate": {
          "type": "BlockStatement",
          "body": [
            {
              "type": "ExpressionStatement",
              "expression": {
                "type": "Literal",
                "value": 5,
                "raw": "5"
              }
            }
          ]
        }
      }
    }
  ]
}
